### PR TITLE
Add module map

### PR DESCRIPTION
**Pull request by dnlkoch**
Add the modules Map and OverviewMap including some default beans. 

This work is heavily inspired by work done by @ahennr! Thx!

### CODE REVIEW

File: src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/Map.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	/**
+	 *
+	 */
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * The MapConfig used by this Map. A MapConfig can be used by several maps
+	 * or overview maps.
+	 */
+	@OneToOne(
```
Comment (buehner) [inline src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/Map.java]: According to the JavaDoc above:

*A MapConfig can be used by several maps or overview maps.*,

shouldn't this be a `@ManyToOne` relation?

File: src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/Map.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+			fetch = FetchType.EAGER,
+			cascade = CascadeType.ALL
+	)
+	private MapConfig mapConfig;
+
+	/**
+	 *
+	 */
+	@OneToOne(cascade = CascadeType.ALL)
```
Comment (buehner) [inline src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/Map.java]: Shouldn't this be a `@ManyToOne` relation, too?
I think it should be possible to use the same `OverviewMap` in multiple `Map`s.

File: src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/Map.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+	/* (non-Javadoc)
+	 * @see java.lang.Object#toString()
+	 */
+	@Override
+	public String toString() {
+		return ToStringBuilder.reflectionToString(this, ToStringStyle.DEFAULT_STYLE);
+	}
+
+}
```
Comment (buehner) [inline src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/Map.java]: No newline at end of file :smile:

File: src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/OverviewMap.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+import de.terrestris.shogun2.model.PersistentObject;
+import de.terrestris.shogun2.model.map.MapConfig;
+
+/**
+ *
+ * Class represents a GeoExt.component.OverviewMap, that displays an overview
+ * map of an parent map.
```
Comment (buehner) [inline src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/OverviewMap.java]: What about the parent map? It does not appear in this model!?

File: src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/OverviewMap.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+	/* (non-Javadoc)
+	 * @see java.lang.Object#toString()
+	 */
+	@Override
+	public String toString() {
+		return ToStringBuilder.reflectionToString(this, ToStringStyle.DEFAULT_STYLE);
+	}
+
+}
```
Comment (buehner) [inline src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/OverviewMap.java]: No newline at end of file :smile:

**Comment by marcjansen** [comment]
Once the comments from @buehner are adressed, this is good to go IMO. Thanks!

File: src/shogun2-webapp-archetype/src/main/resources/archetype-resources/src/main/resources/META-INF/spring/__artifactId__-context-initialize-beans.xml [hunk 6]
```diff
@@ -71,6 +69,133 @@
...
+                <ref bean="res17" />
+                <ref bean="res18" />
+                <ref bean="res19" />
+            </util:list>
+        </property>
+        <property name="extent">
+            <bean class="de.terrestris.shogun2.model.layer.util.Extent">
+                <property name="lowerLeft">
+                    <bean class="java.awt.geom.Point2D$Double">
+                        <constructor-arg index="0" value="-180.0" />
```
Comment (weskamm) [inline src/shogun2-webapp-archetype/src/main/resources/archetype-resources/src/main/resources/META-INF/spring/__artifactId__-context-initialize-beans.xml]: Is it ok to use geographic coords on a metric projection at this stage?

**Comment by dnlkoch** [comment]
Thx for the feedback everyone!

I've adjusted all aspects addressed and if it looks fine to you, feel free to merge.

**Comment by buehner** [comment]
I think we should discuss the Map/OverviewMap relation tomorrow... (WIP)

**Comment by dnlkoch** [comment]
To be more in sync with the BasiGX package the overviewMap module relation in the Map module was removed and as the overviewMap can be used by several maps the `@ManyToOne` seems to be fine.

In addition the overviewMapConfig was removed as these values aren't used by the client as well. Using the `@JsonIdentityInfo` and `@JsonIdentityReference` annotations the passed column of the parentMap (e.g. _name_) is serialized by jackson only.

**Comment by buehner** [comment]
LGTM, i will merge. thx

**Pull request merged**